api/src/services/collections.ts

```diff
@@ -338,25 +344,27 @@ export class CollectionsService {
 			throw new ForbiddenException();
 		}
 
-		await this.knex.transaction(async (trx) => {
-			const service = new CollectionsService({
-				schema: this.schema,
-				accountability: this.accountability,
-				knex: trx,
+		try {
+			await this.knex.transaction(async (trx) => {
+				const service = new CollectionsService({
+					schema: this.schema,
+					accountability: this.accountability,
+					knex: trx,
+				});
+
+				for (const collectionKey of collectionKeys) {
+					await service.updateOne(collectionKey, data, { autoPurgeCache: false });
+				}
 			});
 
-			for (const collectionKey of collectionKeys) {
-				await service.updateOne(collectionKey, data, { autoPurgeCache: false });
+			return collectionKeys;
```

Comment:
From here until the end of the file, values are returned inside try block. Although, in functions above, values are returned after finally block. I think we should be consistent and do it always inside try block, since we may want to return some variable inside the try block and only can be accessed inside the try block. Also, the finally block runs before the function returns the value, so we should be good 👍